languages/js/src/dataFiltering.ts

```diff
@@ -119,22 +120,34 @@ function groundFilter(results: Map<number, unknown[]>, filter: Filter): Filter {
   const ref = filter.value;
   if (!(ref instanceof Ref)) return filter;
 
-  let value = results.get(ref.resultId);
-  value = !ref.field ? value : value?.map(v => (v as any)[ref.field!]); // eslint-disable-line @typescript-eslint/no-explicit-any
+  const { field, resultId } = ref;
+  let value = results.get(resultId);
+
+  if (field !== undefined) {
+    value = value?.map(v => {
+      // TODO(gj): what should we do here if `v` isn't indexable? Without
+      // handling the case where `v` can't be indexed by `field`, it'll blow up
+      // at runtime.
```

Comment:
@gw000 any thoughts?

Reply:
Undefined behavior I'd say. If this happens it's because something is wrong, either with the data filtering configuration or with the user's ORM layer or whatever. Blowing up at runtime sounds about right to me.

Reply:
70db50cf